great_expectations/checkpoint/v1_checkpoint.py

```diff
@@ -221,13 +221,31 @@ def _run_actions(
         checkpoint_result: CheckpointResult,
     ) -> None:
         action_context = ActionContext()
-        for action in self.actions:
+        sorted_actions = self._sort_actions()
+        for action in sorted_actions:
             action_result = action.v1_run(
                 checkpoint_result=checkpoint_result,
                 action_context=action_context,
             )
             action_context.update(action=action, action_result=action_result)
 
+    def _sort_actions(self) -> List[CheckpointAction]:
+        """
+        UpdateDataDocsActions are prioritized to run first, followed by all other actions.
+
+        This is due to the fact that certain actions reference data docs sites,
+        which must be updated first.
+        """
+        priority_actions: List[CheckpointAction] = []
+        secondary_actions: List[CheckpointAction] = []
+        for action in self.actions:
+            if isinstance(action, UpdateDataDocsAction):
+                priority_actions.append(action)
+            else:
+                secondary_actions.append(action)
+
+        return priority_actions + secondary_actions
```

Comment:
We should be able to embedd this into the `CheckpointActions` themselves.
And then you can just call `sorted(my_list)` or `my_list.sort()`
https://docs.python.org/3/reference/datamodel.html#object.__gt__
https://docs.python.org/3/reference/datamodel.html#object.__lt__

https://docs.python.org/3/howto/sorting.html#sorting-basics

Reply:
@Kilo59 in discussing this will @billdirks further, I think we've decided to defer such efforts until they are actually required. In the case we need to deal with dependency graphs when it comes to these actions, we should determine a more robust solution than the naive sorting solution I previously wrote.

Appreciate you chiming in here though!

Reply:
We are keeping sorting internal for now. There is only 1 action that needs to run before everything else. It's unclear if we really want to sort in general or what the requirements are (eg do we really need a dependency graph and is it really better to state your dependencies explicitly vs a sort priority). So I want to punt on proper sorting for now and anything that touches our public api (we expect people to write more actions).